token-allocation/src/types.ts

```diff
@@ -1,6 +1,29 @@
-export type Allocation = {
+import { BigNumberish } from 'ethers/utils'
+
+export type TokenAllocation = {
   symbol: string
-  balance: number
+  balance: BigNumberish
   decimals: number
+  price?: number
+  currency?: string
+}
+
+export type TokenAllocations = TokenAllocation[]
+
+type QuoteSymbol = {
+  price?: number
+  marketCap?: number
+}
+
+type Quote = {
+  [symbol: string]: QuoteSymbol
+}
+
+type ResponseAsset = {
+  quote: Quote
+  balance: string
+}
+
+export type Response = {
+  [symbol: string]: ResponseAsset
```

Comment:
No need to name all these intermediate types. Naming is hard.

```suggestion
export type Response = {
  [symbol: string]: {
    quote: {
      [symbol: string]: {
        price?: number
        marketCap?: number
      }
    }
    balance: string
  }
```